### DISCUSSION TITLE
How to access nvim's clipboard in remote ssh?

### DISCUSSION THREAD

**Discussion by ChenZhongPu** [LazyVim/LazyVim#1565] [Q&A] [answered]
I know this problem is not directly related with lazyvim, but I would like to hear suggestion from other guys.

I am using a client (A) to connect a remote machine (B) with ssh, and both are running `tmux`. In B, I opened a file using nvim, and the `checkhealth` reports that:

>     Clipboard (optional)
>         OK Clipboard tool found: tmux

I would like to copy something in B and then paste it in A. Currently, it can be achieved by `tmux`'s built-in selection and copying. 

But I would like to copy some contents using regular yanking with `y` in nvim, and then paste them in A. How can I achieve this?

**Comment by dpetka2001** [comment, marked as answer]
Never used this before, but I found a pretty recent [write-up](https://gronskiy.com/posts/2023-03-26-copy-via-vim-tmux-ssh/) that you might want to check in case it helps you do what you want.

**Reply by ChenZhongPu** [reply]
Thanks. I tried the method introduced in that blog, and it works!

**Reply by rbgarga** [reply]
I tried that and unfortunately it doesn't work.  Is it still working for you?

**Reply by ChenZhongPu** [reply]
Now I use the tmux's `copy mode` without any extra plugins.

**Reply by rbgarga** [reply]
I've solved it by adding this conf to lazyvim

```lua
vim.o.clipboard = "unnamedplus"
vim.api.nvim_create_autocmd("TextYankPost", {
  callback = function()
    -- vim.highlight.on_yank()
    local copy_to_unnamedplus = require("vim.ui.clipboard.osc52").copy("+")
    copy_to_unnamedplus(vim.v.event.regcontents)
    local copy_to_unnamed = require("vim.ui.clipboard.osc52").copy("*")
    copy_to_unnamed(vim.v.event.regcontents)
  end,
})
```

**Reply by Nukiver** [reply]
> I tried that and unfortunately it doesn't work. Is it still working for you?

The write up worked for me just today.

**Reply by slimctl** [reply]
> I've solved it by adding this conf to lazyvim
> 
> ```lua
> vim.o.clipboard = "unnamedplus"
> vim.api.nvim_create_autocmd("TextYankPost", {
>   callback = function()
>     -- vim.highlight.on_yank()
>     local copy_to_unnamedplus = require("vim.ui.clipboard.osc52").copy("+")
>     copy_to_unnamedplus(vim.v.event.regcontents)
>     local copy_to_unnamed = require("vim.ui.clipboard.osc52").copy("*")
>     copy_to_unnamed(vim.v.event.regcontents)
>   end,
> })
> ```

This worked for me thanks!

**Reply by srbhp** [reply]
> ```lua
> vim.o.clipboard = "unnamedplus"
> vim.api.nvim_create_autocmd("TextYankPost", {
>   callback = function()
>     -- vim.highlight.on_yank()
>     local copy_to_unnamedplus = require("vim.ui.clipboard.osc52").copy("+")
>     copy_to_unnamedplus(vim.v.event.regcontents)
>     local copy_to_unnamed = require("vim.ui.clipboard.osc52").copy("*")
>     copy_to_unnamed(vim.v.event.regcontents)
>   end,
> })
> ```

Woow. this worked like a charm. Thanks a lot.

**Comment by michael-nhat** [comment]
Just notice anyone search and come here, maybe nvim take over mouse feature of tmux or terminal, I can't find way to copy when edit file by neovim in ssh. End up with worked command disable mouse feature in nvim and I can copy as normal: `set mouse=`. Hope that help anyone.
